server/utils/utils.go

```diff
@@ -0,0 +1,350 @@
+package utils
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strings"
+
+	html "github.com/levigross/exp-html"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-confluence/server/config"
+)
+
+const (
+	Script              = "script"
+	Style               = "style"
+	ErrorStatusNotFound = "No content found"
+	Someone             = "Someone"
+)
+
+// GetKeyHash can be used to create a hash from a string
+func GetKeyHash(key string) string {
+	hash := sha256.New()
+	_, _ = hash.Write([]byte(key))
+	return base64.StdEncoding.EncodeToString(hash.Sum(nil))
+}
+
+func Min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func SplitArgs(s string) ([]string, error) {
+	indexes := regexp.MustCompile("\"").FindAllStringIndex(s, -1)
+	if len(indexes)%2 != 0 {
+		return []string{}, errors.New("quotes not closed")
+	}
+
+	indexes = append([][]int{{0, 0}}, indexes...)
+
+	if indexes[len(indexes)-1][1] < len(s) {
+		indexes = append(indexes, [][]int{{len(s), 0}}...)
+	}
+
+	var args []string
+	for i := 0; i < len(indexes)-1; i++ {
+		start := indexes[i][1]
+		end := Min(len(s), indexes[i+1][0])
+
+		if i%2 == 0 {
+			args = append(args, strings.Split(strings.Trim(s[start:end], " "), " ")...)
+		} else {
+			args = append(args, s[start:end])
+		}
+	}
+
+	cleanedArgs := make([]string, len(args))
+	count := 0
+
+	for _, arg := range args {
+		if arg != "" {
+			cleanedArgs[count] = strings.TrimSpace(arg)
+			count++
+		}
+	}
+
+	return cleanedArgs[0:count], nil
+}
+
+func GetPluginKey() string {
```

Comment:
These Plugin methods should be in the plugins file since these are used for the plugins IMHO.

Utils files are anti-patterns we could try to avoid since it's not obvious what these files try to solve, and they usually can be added to files where they're used and have more context.